packages/sanes-chrome-extension/src/extension/background/createPersona/requestCallback.ts

```diff
@@ -0,0 +1,69 @@
+import { PublicIdentity, UnsignedTransaction } from '@iov/bcp';
+import { updateExtensionBadge } from './requestExtensionBadge';
+import { RequestHandler } from './requestHandler';
+import { SenderWhitelist } from './requestSenderWhitelist';
+
+export interface RequestMeta {
+  readonly senderUrl: string;
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function isRequestMeta(data: unknown): data is RequestMeta {
+  if (typeof data !== 'object' || data === null) {
+    return false;
+  }
+
+  return typeof (data as RequestMeta).senderUrl === 'string';
+}
+
+type CallbackResponse = ReadonlyArray<PublicIdentity> | boolean;
+
+async function requestCallback(
+  reason: string,
+  meta: RequestMeta,
```

Comment:
this type is wrong: since you did not yet check what kind of data you got here, this is `any`. A couple of lines below, you have the check that ensures `meta` is the type you want.